### ISSUE TITLE
mehalter Major Cleanup. what it means

### ISSUE THREAD

**Issue by jayp0521** [AstroNvim/AstroNvim#423]
hello,

I know you are the creator of this repo, and I follow your repo for examples on some custom additions. I am a big fan of this project. 

I preferred the approach of not knowing the correct keys and separation of files etc. Basically, I am wondering if you are planning on dropping support for the repo to work without an `init.lua` since using that seems to require tribal knowledge of the repo. I am not a fan of that approach. are you planning on keeping support for these sorts of options: https://github.com/AstroNvim/AstroNvim/blob/7501395dc11a6d431fa4f37de72f61ab371a6c1d/lua/configs/lsp/init.lua#L30?

https://git.mehalter.com/mehalter/AstroNvim_user/commit/ff8c7a3df7763c77c31a8faa6d0c2742885e7b99

**Comment by jayp0521** [comment]
https://astronvim.github.io/recipes/black_belt#:~:text=mehalter%20%2D%20Fully%20separated%20into%20individual%20files%2C%20shows%20that%20the%20init.lua%20file%20is%20not%20necessary...

also should probably update this :) lol

**Comment by mehalter** [comment]
@jayp0521 Thanks for reminding me to update that line in the docs! Also no we are not planning on changing anything about how configuration is supported or anything don't worry! I was just moving some stuff around in my set up.

**Comment by mehalter** [comment]
I think I have re-read and think I have misinterpreted. Are you saying you "dislike this feature"? 😂 Sorry if I misinterpreted!

**Comment by mehalter** [comment]
also you mentioned that the automatic file separation approach requires tribal knowledge of the repo and the correct keys. The main goal of how we structured this approach was to make it easy for the file structure to be implied by the init.lua table structure so you wouldn't need tribal knowledge. Basically the automatic filenstructure that is searched is the exact same as the nested table structure of the init.lua. Let ke know if I explained this incorrectly or unclear!
